lib/pure/strutils.nim

```diff
@@ -1442,30 +1512,151 @@ proc formatFloat*(f: float, format: FloatFormatMode = ffDefault,
+  ## See documentation under forward declaration (above)
+  var
+    absolute: BiggestFloat
+    significand: BiggestFloat
+    fexponent: BiggestFloat
+    exponent: int
+    splitResult: seq[string]
+    suffix: string = ""
+  proc getPrefix(exp: int): char =
+    ## Get the SI prefix for a given exponent
+    ##
+    ## Assumes exponent is a multiple of 3; returns ' ' if no prefix found
+    let siPrefixes = ['a','f','p','n','u','m',' ','k','M','G','T','P','E']
+    var index: int = (exp div 3) + 6
+    result = ' '
+    if index in low(siPrefixes)..high(siPrefixes):
+      result = siPrefixes[index]
+
+  # Most of the work is done with the sign ignored, so get the absolute value
+  absolute = abs(f)
+  significand = f
+
+  if absolute == 0.0:
+    # Simple case: just format it and force the exponent to 0
+    exponent = 0
+    result = significand.formatBiggestFloat(ffDecimal, precision)
+  else:
+    # Find the best exponent that's a multiple of 3
+    fexponent = round(floor(log10(absolute)))
+    fexponent = 3.0 * round(floor(fexponent / 3.0))
+    # Adjust the significand for the new exponent
+    significand /= pow(10.0, fexponent)
+
+    # Round the significand and check whether it has affected
+    # the exponent
+    significand = round(significand, precision)
+    absolute = abs(significand)
+    if absolute >= 1000.0:
+      significand /= 1000.0
+      fexponent += 3
+    # Components of the result:
+    result = significand.formatBiggestFloat(ffDecimal, precision)
+    exponent = fexponent.int()
+
+  let decimalCharacters = {'.', ','}
+  splitResult = result.split(decimalCharacters)
+  result = splitResult[0]
+  # result should have at most one decimal character
+  if splitResult.len() > 1:
+    # If trim is set, we get rid of trailing zeros
+    if trim:
+        while splitResult[1].endsWith("0"):
+          # Trim last character (ugly version of python's `x = x[:-1]`)
+          splitResult[1] = splitResult[1].substr(0, splitResult[1].len()-2)
+        if splitResult[1].len() > 0:
+          result &= decimalSep & splitResult[1]
+    else:
+      result &= decimalSep & splitResult[1]
+
+  # Combine the results accordingly
+  if exponent == 0:
+    if unit != nil:
+      suffix = " " & unit
+  else:
+    if siPrefix:
+      var p = getPrefix(exponent)
+      if p != ' ':
+        suffix = " " & p
+    if suffix == "":
+      suffix = "e" & $exponent
+    if unit != nil:
+      suffix &= unit
+  result &= suffix
+
+  for c in decimalCharacters:
+    result = result.replace(c, decimalSep)
 
-proc formatSize*(bytes: BiggestInt, decimalSep = '.'): string =
-  ## Rounds and formats `bytes`. Examples:
+type
+  BinaryPrefixMode* = enum ## the different names for binary prefixes
+    bpIEC, # use the IEC/ISO standard prefixes such as kibi
+    bpColloquial # use the colloquial kilo, mega etc
+
+proc formatSize*(bytes: int64,
+                 decimalSep = '.',
+                 prefix = bpIEC,
```

Comment:
The lack of backwards compatibility was deliberate (as it made no sense at all in my opinion with 2.234GB meaning 2 GB **or** 2340 MB rather than something more logical like 2.234 GB or 2 GB **and** 234 MB).  I'll chance includeSpace.